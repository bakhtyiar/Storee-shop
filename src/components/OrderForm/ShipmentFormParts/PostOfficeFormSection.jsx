import React from 'react';
import FormTextField from "../../formikElements/FormTextField";
import {Col, Row} from "react-bootstrap";
import FormSelectField from "../../formikElements/FormSelectField";
import {states} from "../../../utils/constants";

const PostOfficeFormSection = () => {
    return (
        <Row data-testid="post-office-form-section">
            <FormTextField
                as={Col}
                controlId="formGridCity"
                label='City'
                placeholder='New York'
                name='city'
                type='text'
            />
            <FormSelectField
                as={Col}
                controlId="formGridState"
                label='State'
                placeholder='New York'
                name='state'
                type='text'
            >
                <>
                    <option value={''} disabled>Choose...</option>
                    {states.map((state) => (
                        <option key={state.value}
                                value={state.value}>{state.label}</option>
                    ))}
                </>
            </FormSelectField>
            <FormTextField
                as={Col}
                controlId="formGridZip"
                label='Zip'
                placeholder='10000'
                name='postalCode'
                type='text'
            />
        </Row>
    );
};

export default PostOfficeFormSection;