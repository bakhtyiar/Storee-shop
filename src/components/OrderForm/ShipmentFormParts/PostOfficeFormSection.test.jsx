import {screen} from "@testing-library/react";
import {renderWithFormikContext} from "../../../tests/helpers/renderWithFormikContext";
import PostOfficeFormSection from "./PostOfficeFormSection";

describe("Basic tests", () => {
    test("Match snapshot", () => {
        renderWithFormikContext(<PostOfficeFormSection/>);
        expect(screen.getByTestId("post-office-form-section")).toMatchSnapshot();
    })
})