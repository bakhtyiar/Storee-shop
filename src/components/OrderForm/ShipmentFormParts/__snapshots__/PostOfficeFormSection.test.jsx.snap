// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Match snapshot 1`] = `
<div
  class="row"
  data-testid="post-office-form-section"
>
  <div
    class="col"
    data-testid="form-text-field"
  >
    <label
      class="form-label"
      for="formGridCity"
    >
      City
    </label>
    <div
      class="input-group"
    >
      <input
        class="rounded form-control"
        id="formGridCity"
        name="city"
        placeholder="New York"
        type="text"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
  </div>
  <div
    class="col"
    data-testid="form-select-field"
  >
    <label
      class="form-label"
      for="formGridState"
    >
      State
    </label>
    <div
      class="input-group"
    >
      <select
        as="select"
        class="rounded form-select"
        id="formGridState"
        name="state"
        type="text"
      >
        <option
          disabled=""
          value=""
        >
          Choose...
        </option>
        <option
          value="AL"
        >
          Alabama (AL)
        </option>
        <option
          value="AK"
        >
          Alaska (AK)
        </option>
        <option
          value="AZ"
        >
          Arizona (AZ)
        </option>
        <option
          value="AR"
        >
          Arkansas (AR)
        </option>
        <option
          value="CA"
        >
          California (CA)
        </option>
        <option
          value="CO"
        >
          Colorado (CO)
        </option>
        <option
          value="CT"
        >
          Connecticut (CT)
        </option>
        <option
          value="DE"
        >
          Delaware (DE)
        </option>
        <option
          value="FL"
        >
          Florida (FL)
        </option>
        <option
          value="GA"
        >
          Georgia (GA)
        </option>
        <option
          value="HI"
        >
          Hawaii (HI)
        </option>
        <option
          value="ID"
        >
          Idaho (ID)
        </option>
        <option
          value="IL"
        >
          Illinois (IL)
        </option>
        <option
          value="IN"
        >
          Indiana (IN)
        </option>
        <option
          value="IA"
        >
          Iowa (IA)
        </option>
        <option
          value="KS"
        >
          Kansas (KS)
        </option>
        <option
          value="KY"
        >
          Kentucky (KY)
        </option>
        <option
          value="LA"
        >
          Louisiana (LA)
        </option>
        <option
          value="ME"
        >
          Maine (ME)
        </option>
        <option
          value="MD"
        >
          Maryland (MD)
        </option>
        <option
          value="MA"
        >
          Massachusetts (MA)
        </option>
        <option
          value="MI"
        >
          Michigan (MI)
        </option>
        <option
          value="MN"
        >
          Minnesota (MN)
        </option>
        <option
          value="MS"
        >
          Mississippi (MS)
        </option>
        <option
          value="MO"
        >
          Missouri (MO)
        </option>
        <option
          value="MT"
        >
          Montana (MT)
        </option>
        <option
          value="NE"
        >
          Nebraska (NE)
        </option>
        <option
          value="NV"
        >
          Nevada (NV)
        </option>
        <option
          value="NH"
        >
          New Hampshire (NH)
        </option>
        <option
          value="NJ"
        >
          New Jersey (NJ)
        </option>
        <option
          value="NM"
        >
          New Mexico (NM)
        </option>
        <option
          value="NY"
        >
          New York (NY)
        </option>
        <option
          value="NC"
        >
          North Carolina (NC)
        </option>
        <option
          value="ND"
        >
          North Dakota (ND)
        </option>
        <option
          value="OH"
        >
          Ohio (OH)
        </option>
        <option
          value="OK"
        >
          Oklahoma (OK)
        </option>
        <option
          value="OR"
        >
          Oregon (OR)
        </option>
        <option
          value="PA"
        >
          Pennsylvania (PA)
        </option>
        <option
          value="RI"
        >
          Rhode Island (RI)
        </option>
        <option
          value="SC"
        >
          South Carolina (SC)
        </option>
        <option
          value="SD"
        >
          South Dakota (SD)
        </option>
        <option
          value="TN"
        >
          Tennessee (TN)
        </option>
        <option
          value="TX"
        >
          Texas (TX)
        </option>
        <option
          value="UT"
        >
          Utah (UT)
        </option>
        <option
          value="VT"
        >
          Vermont (VT)
        </option>
        <option
          value="VA"
        >
          Virginia (VA)
        </option>
        <option
          value="WA"
        >
          Washington (WA)
        </option>
        <option
          value="WV"
        >
          West Virginia (WV)
        </option>
        <option
          value="WI"
        >
          Wisconsin (WI)
        </option>
        <option
          value="WY"
        >
          Wyoming (WY)
        </option>
      </select>
      <div
        class="invalid-feedback"
      />
    </div>
  </div>
  <div
    class="col"
    data-testid="form-text-field"
  >
    <label
      class="form-label"
      for="formGridZip"
    >
      Zip
    </label>
    <div
      class="input-group"
    >
      <input
        class="rounded form-control"
        id="formGridZip"
        name="postalCode"
        placeholder="10000"
        type="text"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
  </div>
</div>
`;
